float32 movement_scaling_factor
---
bool success
